### DISCUSSION TITLE
Forwarder but filter to NOT forward certain messages.

### DISCUSSION THREAD

**Discussion by meta-meta** [djipco/webmidi#346] [Q&A] [answered]
Hi, I'm looking for a way to `addForwarder` but I'd like to consume certain messages in my app, and forward anything that has not been consumed. Is this possible?

**Comment by djipco** [comment, marked as answer]
There is no filtering mechanism in the `Forwarder` object. However, you could create your own [`Forwarder`](https://webmidijs.org/api/classes/Forwarder) object and use its [`forward()`](https://webmidijs.org/api/classes/Forwarder#forward) method when it's appropriate to your use case.

**Reply by meta-meta** [reply]
Oh, that's perfect. Thanks!
